pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -416,7 +416,7 @@ pub mod pallet {
 				return Err(Error::<T>::InvalidMessageOrigin.into());
 			}
 
-			Self::receive(router_id, origin_address, msg.into())
+			Self::receive(router_id, origin_address, T::Message::deserialize(&msg)?)
```

Comment:
I think we need to move this out from the gateway. If not, the caller of this extrinsic will skip the forwarding logic.

Still not sure where, maybe on `pallet-axelar-router` or `pallet-liquidity-pool-forwarder` (and using `Message` instead of `Vec<u8>`).

Or maybe just removing it, because it's no longer used...

Reply:
Great catch! I propose to add a `MessageDeserializer` AT to the Gateway config which is implemented by our runtime type `MessageSerializer`. This will deserialize the message and pass it to the forwarder pallet which will pass it back to the `MessageReiver` impl of the Gateway.

So from now:
* `process` extrinsic -> Gateway impl of `MessageReceiver`

To
* `process` in Gateway -> `receive` in `MessagSerializer` which deserializes -> `receive` in forward pallet which unwraps any forwarded message -> `receive` in Gateway which does its thing

WDYT?

Reply:
This requires instantiating the `router_message` mock pallet, which I am working on right now.

Reply:
Mmm... I am not sure about that walk...

As more I think on it as more I think we can remove this. We no longer need this since Axelar uses the trait to send message to the gateway. If we want to emulate receiving message explicitly, this extrinsic should be in axelar-router

Reply:
Interested in your take on this @cdamian

Reply:
See here for code changes if we added the `MessageDeserializer` to the Gateway: [a253a6f](https://github.com/centrifuge/centrifuge-chain/pull/1971/commits/a253a6fcb3db9292ca5686d2df651fd074d345a0)

However, I have reverted that change and applied @lemunozm's proposal of removing the `receive_message` extrinsic which apparently was only used by the Axelar precompile. I have refactored the existing Gateway UTs for checking the `MessageReceiver` implementation which was not tested: [cbb75b4](https://github.com/centrifuge/centrifuge-chain/pull/1971/commits/cbb75b4605c618a32885be53259db8319ce49b40)